sthree/bucket.go

```diff
@@ -656,7 +660,8 @@ func (b *Bucket) SyncTo(local, prefix string, opts SyncOptions) error {
 		return nil
 	}))
 
-	ctx, cancel := context.WithTimeout(context.TODO(), opts.Timeout)
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, opts.Timeout)
```

Comment:
Do you need to handle the case where `opts.Timeout` is `0`?

Reply:
I don't think so. I think these are effectively constants in almost every case, and in the case where the timeout is to short (e.g. 0), it'll just error quickly, which is about all validation of the timeout value in opts would give us.